src/server/repositories/UrlRepository.ts

```diff
@@ -54,28 +63,58 @@ export class UrlRepository implements UrlRepositoryInterface {
   }
 
   public create: (
-    properties: { userId: number; shortUrl: string; longUrl?: string },
+    properties: {
+      userId: number
+      shortUrl: string
+      longUrl?: string
+      tags?: string[]
+    },
     file?: StorableFile,
   ) => Promise<StorableUrl> = async (properties, file) => {
     const newUrl = await sequelize.transaction(async (t) => {
-      await Url.create(
-        {
-          ...properties,
-          longUrl: file
-            ? this.fileBucket.buildFileLongUrl(file.key)
-            : properties.longUrl,
-          isFile: !!file,
-        },
-        {
-          transaction: t,
-        },
-      )
+      const tagStrings = properties.tags
+        ? properties.tags.join(tagSeparator)
+        : ''
+      const urlStaticDTO = {
+        ...properties,
+        longUrl: file
+          ? this.fileBucket.buildFileLongUrl(file.key)
+          : properties.longUrl,
+        isFile: !!file,
+        tagStrings,
+      }
+      const url = await Url.create(urlStaticDTO, {
+        transaction: t,
+      })
+      if (properties.tags) {
+        const tagCreationResponses = await Promise.all(
```

Comment:
Should we use the new upsert function here?

Reply:
agreed, done